packages/loaders/src/styled/StyledSkeleton.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import styled, { keyframes, css, ThemeProps, DefaultTheme } from 'styled-components';
+import rgba from 'polished/lib/color/rgba';
+import { DEFAULT_THEME, retrieveComponentStyles, getColor } from '@zendeskgarden/react-theming';
+
+const COMPONENT_ID = 'loaders.skeleton';
+
+const fadeInAnimation = keyframes`
+  0%, 60% {
+    opacity: 0;
+  }
+
+  100% {
+    opacity: 1;
+  }
+`;
+
+const skeletonAnimation = keyframes`
+  100% {
+    left: 100%;
+  }
+`;
+
+const skeletonRtlAnimation = keyframes`
+  100% {
+    right: 100%;
+  }
+`;
+
+const retrieveSkeletonBackgroundColor = ({
+  theme,
+  dark
+}: IStyledSkeletonProps & ThemeProps<DefaultTheme>) => {
+  if (dark) {
+    return css`
+      background-color: ${rgba(theme.colors.background, 0.2)};
+    `;
+  }
+
+  return css`
+    background-color: ${getColor('neutralHue', 800, theme, 0.1)};
+  `;
+};
+
+interface IStyledSkeletonProps {
+  width?: string;
+  height?: string;
+  dark?: boolean;
+  customWidth?: string;
+  customHeight?: string;
+}
+
+const retrieveSkeletonAnimation = ({ theme }: ThemeProps<DefaultTheme>) => {
+  if (theme.rtl) {
+    return css`
+      right: -1800px;
+      animation: ${skeletonRtlAnimation} 1.5s ease-in-out 300ms infinite;
+    `;
+  }
+
+  return css`
+    left: -1800px;
+    animation: ${skeletonAnimation} 1.5s ease-in-out 300ms infinite;
+  `;
+};
+
+const retrieveSkeletonGradient = ({
+  theme,
+  dark
+}: IStyledSkeletonProps & ThemeProps<DefaultTheme>) => {
+  // Disabling stylelint due to conflicts with prettier and linear-gradient formatting
+  return css`
+    /* stylelint-disable */
+    background-image: linear-gradient(
+      ${theme.rtl ? '-45deg' : '45deg'},
+      transparent,
+      ${dark ? getColor('kale', 700, theme, 0.4) : rgba(theme.colors.background, 0.6)},
+      transparent
+    );
+    /* stylelint-enable */
+  `;
+};
+
+export const StyledSkeleton = styled.div.attrs({
+  'data-garden-id': COMPONENT_ID,
+  'data-garden-version': PACKAGE_VERSION
+})<IStyledSkeletonProps>`
+  display: inline-block;
+  position: relative;
+  animation: ${fadeInAnimation} 750ms linear;
+  border-radius: ${props => props.theme.space.base}px;
+  width: ${props => props.customWidth};
+  height: ${props => props.customHeight};
+  overflow: hidden;
+  line-height: 0.6;
```

Comment:
Line height is typically a function of intended pixel height divided by font-size. How may that be represented here?

Reply:
The `Skeleton` component is meant to fill any element it is put in. Since both of those values will always be inherited is there a way to calculate this?

Reply:
There may not be. Although, usually our designers identify values with some basis in mind and it is nice when we can capture that intent in code. For example, we frequently see 14px fonts occupying 20px of space. Therefore the (general) intent is a 1.42857 line-height, regardless of font size. (Side note: these unitless values tend to lose their effectiveness at extreme sizes.)

Reply:
I now calculate this based on the `fontSize` and `lineHeights` provided by the theme.